apps/doc-gen/src/app/docgen-client.ts

```diff
@@ -45,19 +48,22 @@ export class DocGenClient extends PluginClient {
         input: input,
         output: output
       }
-      this.fileName = fileName
-      this.eventEmitter.emit('compilationFinished', this.build, fileName)
+      const segmentedPathList = normalizeContractPath(fileName)
+      this.fileName = segmentedPathList[segmentedPathList.length - 1]
+      this.contractPath =  segmentedPathList[0]
+      this.eventEmitter.emit('compilationFinished', this.build, this.fileName)
     })
   }
 
   async docgen(builds: Build[], userConfig?: Config): Promise<void> {
     const config = { ...defaults, ...userConfig }
+    config.sourcesDir = this.contractPath.length > 0 ? this.contractPath : config.sourcesDir
```

Comment:
it seems `this.contractPath.length` will always be `> 0`, in which case do you think it isn't?

Reply:
Ah..you are right.